mmdet/models/builder.py

```diff
@@ -55,5 +60,21 @@ def build_detector(cfg, train_cfg=None, test_cfg=None):
         'train_cfg specified in both outer field and model field '
     assert cfg.get('test_cfg') is None or test_cfg is None, \
         'test_cfg specified in both outer field and model field '
-    return DETECTORS.build(
+
+    rfsearch_cfg = cfg.pop('rfsearch_cfg', None)
+
+    detector = DETECTORS.build(
         cfg, default_args=dict(train_cfg=train_cfg, test_cfg=test_cfg))
+    if rfsearch_cfg is not None:
+        assert RFSearchHook is not None, 'Please install mmcv > 1.7.0'
+        rfsearch_warp = RFSearchHook(
+            mode=rfsearch_cfg.get('mode', 'search'),
+            config=rfsearch_cfg.get('config', None),
+            rfstructure_file=rfsearch_cfg.get('rfstructure_file', None),
+            by_epoch=rfsearch_cfg.get('by_epoch', True),
+            verbose=rfsearch_cfg.get('verbose', True),
+        )
+        detector.init_weights()
```

Comment:
I think it's better to put this part into train.py. In this way, we can only define `RFSearchHook` in config and delete the model `rfnext_cfg`.

BTW, it's better to separate the function of `init_model` from `RFSearchHook`. So that we don't need to build redundant `RFSearchHook` in train.py.

Reply:
By the way, init_model should be called before constructing optimizer, because there will be new parameters after calling warp_model.

Reply:
Should I modify the codes of mmcv again, to separate the function of `init_model` from `RFSearchHook`?